File: 31588304_1.py
import requests
import re
import time

from bs4 import BeautifulSoup
from pymongo import MongoClient
client = MongoClient("mongodb://localhost:27017/")

dblist=client.list_database_names()
for sd in dblist:
    print(str(sd),end=", ")

db=client["stockBase"]
print(str(db))

dblist=client.list_database_names()
for sd in dblist:
    print(str(sd),end=", ")

col=db["Stocks"]
while(1):
    r = requests.get('https://finance.yahoo.com/most-active')
    rsc=r.status_code
    if rsc>=300 or rsc<=199:
        print("Can't connect. Retrying:")
        print(rsc)
        r=requests.get('https://finance.yahoo.com/most-active')
        rsc=r.status_code
        if rsc>=300 or rsc<=199:
            print("Retry failed")
            print(rsc)
            break
    print("Connection successful")
    soup1=BeautifulSoup(r.text, "lxml")
    h=soup1.html
    print(h.head.title.text)
    base=h.tbody
    #print("Name: "+base.name+"\tAttr: "+str(base.attrs))
    index=0
    for s in base.contents:
        symbol=s.contents[0].text
        print(symbol, end=", ")
        name=s.contents[1].text
        print(name, end=", ")
        price=float(s.contents[2].text)
        print(price, end=", ")
        change=float(s.contents[3].text)
        print(change, end=", ")
        vol=(s.contents[5].text)
        print(vol)
        if((str(col.find_one({"_id":index})))=="None"):
            col.insert_one({"_id":index, "Symbol":symbol, "Name":name, "Price":price, "Change":change, "Volume":vol})
        else:
            col.delete_one({"_id":index})
            col.insert_one({"_id":index, "Symbol":symbol, "Name":name, "Price":price, "Change":change, "Volume":vol})
        index+=1


    time.sleep(180)

File: 31588304_1.php
<?php
#------------------------------------------------------------------------------------------------------------

require 'vendor/autoload.php';
$client = new MongoDB\Client("mongodb://@localhost:27017/");
$db = $client->stockBase;
$collection = $db->Stocks;

#declares $client as MongoDB server client	|	$db as stockBase database	|	$collection as collection Stocks

$result = $collection->find( [] );
if (!isset($_GET['sor']))
{
	$result=$collection->find([]);
}
if (isset($_GET['sor']))
{
	if ($_GET['sor'] == '_id')
	{
		$result=$collection->find([],['sort'=>['_id'=>1]]);
	}
	elseif ($_GET['sor'] == 'Symbol')
	{
		$result=$collection->find([],['sort'=>['Symbol'=>1]]);
	}
	elseif ($_GET['sor'] == 'Name')
	{
		$result=$collection->find([],['sort'=>['Name'=>1]]);
	}
	elseif ($_GET['sor'] == 'Price')
	{
		$result=$collection->find([],['sort'=>['Price'=>1]]);
	}
	elseif ($_GET['sor'] == 'Change')
	{
		$result=$collection->find([],['sort'=>['Change'=>1]]);
	}
	elseif ($_GET['sor'] == 'Volume')
	{
		$result=$collection->find([],['sort'=>['Volume'=>1]]);
	}
}
#------------------------------------------------------------------------------------------------------------
echo "<table border='1'>\n";
	#table header
	echo "<thead>\n";
		echo "<tr>\n";
			echo "<th><a href=\"?sor=_id\">Index</a></th><th><a href=\"?sor=Symbol\">Symbol</a></th><th><a href=\"?sor=Name\">Name</a></th><th><a href=\"?sor=Price\">Price(Intraday)</a></th><th><a href=\"?sor=Change\">Change</a></th><th><a href=\"?sor=Volume\">Volume</a></th>";
		echo "</tr>\n";
	echo "</thead>\n";

	# table body
	echo "<tbody>\n";
	foreach ($result as $doc) 
	{
		echo "<tr>\n  ";
		echo "<td>{$doc['_id']}</td>";
		echo "<td>{$doc['Symbol']}</td>";
		echo "<td>{$doc['Name']}</td>";
		echo "<td>{$doc['Price']}</td>";
		echo "<td>{$doc['Change']}</td>";
		echo "<td>{$doc['Volume']}</td>";
		/*foreach ($doc as $key => $value)
		{
			echo "<td>";
			echo "${value}";
			echo "</td>";
		}*/
		echo "\n</tr>\n";
	}
	echo "</tbody>\n";
echo "</table>\n";
?>
